05-JWT-Basics/preferred/public/index.js

```diff
@@ -0,0 +1,53 @@
+const form = document.querySelector('.form');
+const submitButton = document.querySelector('.submit-button'); 
+const nameInput = document.querySelector('.name-input');
+const passwordInput = document.querySelector('.password-input');
+const loginMessage = document.querySelector('.login-message'); 
+const authButton = document.querySelector('.auth-button')
+const authMessage = document.querySelector('.auth-message');
+const successMessage = document.querySelector('.success');
+
+loginMessage.hidden = true; 
+successMessage.hidden = true; 
+
+form.addEventListener('submit', async (e) => {
+  e.preventDefault(); 
+  const name = nameInput.value;
+  const password = passwordInput.value; 
+  console.log('submit', name, password); 
+
+  if (!name || !password) {
+    loginMessage.hidden = false;
+  } 
+  if (name && password) {
```

Comment:
You shouldn't actually need this check, since this is the same thing as the above `!name || !password` check. You can have the below lines outright without issue, given adding the `return` in the first `if`:
```js
// ...
}

loginMessage.hidden = true;
successMessage.hidden = false;
nameInput.value = '';
// ...
```